src/CSharp/CodeCracker/Refactoring/XmlDocumentationCreateMissingParametersCodeFixProvider.cs

```diff
@@ -0,0 +1,76 @@
+            var newDocumentationNode = documentationNode;
+            var methodParameterWithDocParameter = GetMethodParametersWithDocParameters(method, documentationNode);
+
+            var xmlText = documentationNode.Content.OfType<XmlTextSyntax>().Skip(1).First();
+
+            var nodesToAdd = methodParameterWithDocParameter.Where(p => p.Item2 == null)
+                                .SelectMany(x => CreateParamenterXmlDocumentation(xmlText, x.Item1.Identifier.ValueText, method.Identifier.ValueText))
+                                .ToList();
+            var newFormation = newDocumentationNode.Content.OfType<XmlElementSyntax>().ToList();
+            var node = newFormation.LastOrDefault(xEle => xEle.StartTag.Name.LocalName.ValueText == "param") ?? newFormation.LastOrDefault(xEle => xEle.StartTag.Name.LocalName.ValueText == "summary");
+            var nodeInList = documentationNode.Content.OfType<XmlTextSyntax>().FirstOrDefault(x => x.FullSpan.Start == node.FullSpan.End);
+
+            newDocumentationNode = newDocumentationNode.InsertNodesAfter(node, nodesToAdd.Cast<SyntaxNode>());
+
+            return root.ReplaceNode(documentationNode, newDocumentationNode);
+        }
+
+
+        private static XmlNodeSyntax[] CreateParamenterXmlDocumentation(XmlTextSyntax xmlText, string paramenterName, string methodName)
+        {
+            var content = $"todo: describe {paramenterName} parameter on {methodName}";
+            var xmlTagName = SyntaxFactory.XmlName(SyntaxFactory.Identifier(@"param "));
+            var startTag = SyntaxFactory.XmlElementStartTag(xmlTagName).WithAttributes(CreateXmlAttributes(paramenterName));
+            var endTag = SyntaxFactory.XmlElementEndTag(SyntaxFactory.XmlName(SyntaxFactory.Identifier(@"param")));
+
+            var xmlElementSyntax = SyntaxFactory.XmlElement(startTag, endTag).WithContent(CreateXmlElementContent(content));
+
+            return new XmlNodeSyntax[] { xmlText, xmlElementSyntax };
+        }
+
+        private static SyntaxList<XmlNodeSyntax> CreateXmlElementContent(string content)
+        {
+            var triviaList = SyntaxFactory.TriviaList();
+            var xmlTextLiteral = SyntaxFactory.XmlTextLiteral(triviaList, content, content, triviaList);
+            return SyntaxFactory.SingletonList<XmlNodeSyntax>(SyntaxFactory.XmlText(SyntaxFactory.TokenList(xmlTextLiteral)));
+        }
+
+        private static SyntaxList<XmlAttributeSyntax> CreateXmlAttributes(string paramenterName)
+        {
+            var syntaxToken = SyntaxFactory.Token(DoubleQuoteToken);
+            var xmlAttributeName = SyntaxFactory.XmlName(SyntaxFactory.Identifier(@"name"));
+
+            return SyntaxFactory.SingletonList<XmlAttributeSyntax>(SyntaxFactory.XmlNameAttribute(xmlAttributeName, syntaxToken, SyntaxFactory.IdentifierName(paramenterName), syntaxToken));
+        }
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
```

Comment:
@sharwell 3 reasons: style, recommendation and performance.
Style: All our `RegisterCodeFixesAsync` are sealed.
Recommendation: this is how Microsoft is recommending on their own template, and this is how it is done on Roslyn itself.
Performance: while the compiler usually emits IL with a `callvirt` to sealed and non-virtual method calls, the method is still marked `final`, which is a hint for the JIT later for possible optimizations, like method inlining.